app/queries/topics_resolver.rb

```diff
@@ -64,10 +69,11 @@ def sort_criterion_string
   end
 
   def filter_by_solution(topics, resolution)
-    topics_with_solution = topics.joins(:posts).where(posts: { solution: true })
+    topics_with_solution =
+      topics.joins(:posts).where(posts: { solution: true }).select(:id)
     case resolution
     when 'Solved'
-      topics_with_solution
+      Topic.where(id: topics_with_solution)
```

Comment:
Why this change? I don't see any functional difference here, except for the nesting of the original data set into a _pick-by-id_ query. This also feels unrelated to this PR.

Reply:
@mahesh-krishnakumar Understood. 👍🏼